calyx-backend/src/firrtl.rs

```diff
@@ -175,12 +152,17 @@ fn get_primitive_module_name(name: &Id, param_binding: &Binding) -> String {
 }
 
 fn emit_primitive_extmodule<F: io::Write>(
+    ports: &SmallVec<[RRC<Port>; 10]>,
```

Comment:
I think this could be just a slice type, to keep it slightly more generic?
```suggestion
    ports: &[RRC<Port>],
```
I assume `SmallVec` supports this "free" slice extraction, but I could be wrong!

Reply:
Yup! And the `&[T]` type is also compatible with references to normal vectors.

Reply:
This is so cool!! I'll read up about slice types, thanks for the suggestion!

Reply:
The relevant `Deref` impl is here: https://docs.rs/smallvec/latest/smallvec/struct.SmallVec.html#impl-Deref-for-SmallVec%3CA%3E